File: backend/src/main/java/com/back/domain/member/member/dto/MemberDto.java
package com.back.domain.member.member.dto;

import com.back.domain.member.member.entity.Member;

public record MemberDto (
    String email,
    String name,
    String password
) {
    public MemberDto (String email, String name, String password) {
        this.email = email;
        this.name = name;
        this.password = password;
    }

    public MemberDto (Member member){
        this(
                member.getEmail(),
                member.getName(),
                member.getPassword()
        );
    }
}

File: backend/src/main/java/com/back/domain/member/member/dto/MemberDto.kt
package com.back.domain.member.member.dto

import com.back.domain.member.member.entity.Member

data class MemberDto(
    val email: String?,
    val name: String?,
    val password: String?
) {
    constructor(member: Member) : this(
        email = member.getEmail(),
        name = member.getName(),
        password = member.getPassword()
    )
}
